microcosm/opaque.py

```diff
@@ -19,15 +19,20 @@
 
 """
 from collections import MutableMapping
-from contextlib import ContextDecorator
+from contextlib import ContextDecorator, ExitStack
 from copy import deepcopy
 from types import MethodType
 
+from opentracing_instrumentation.request_context import span_in_context
+from opentracing.propagation import Format
+from opentracing.ext import tags
+
 
 def _make_initializer(opaque):
 
-    class OpaqueInitializer(ContextDecorator):
+    class OpaqueInitializer(ContextDecorator, ExitStack):
```

Comment:
`ExitStack` makes it easier to chain together a few context manages e.g. the `jaeger` context managers all within opaque initialize.